File: vol-100/01-cubic/cubic.php
<?php

print(strval(cubic(read_num())) . "\n");

function read_num()
{
  if( fscanf(STDIN, "%d", $num) != 1 )
  {
    exit("invalid input");
  }
  return $num;
}

function cubic($num)
{
  return $num * $num * $num;
}

File: vol-100/01-cubic/cubic.pl
use strict;
use warnings;

caller or main();

sub main
{
  print cubic(read_num()), "\n";
}

sub read_num
{
  my $line = <>;
  chomp $line;
  $line =~ /^(\d+)$/ or die "invalid input: $line";
  $1;
}

sub cubic
{
  my $num = shift;
  $num * $num * $num;
}
